controllers/secretproviderclasspodstatus_controller.go

```diff
@@ -266,6 +272,18 @@ func (r *SecretProviderClassPodStatusReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, nil
 	}
 
+	begin := time.Now()
+	providerName := string(spc.Spec.Provider)
+	namespace := spcPodStatus.Namespace
+	secretProviderClass := spc.Name
+	defer func() {
+		// if there is SecretObjects defined in the SPC, then report the metric if sync is successful
+		if e == nil && !res.Requeue {
+			r.reporter.ReportSyncSecretCtMetric(ctx, providerName, namespace, secretProviderClass)
+			r.reporter.ReportSyncSecretDuration(ctx, time.Since(begin).Seconds())
+		}
+	}()
```

Comment:
Given this is only recorded when the error is nil, I would prefer this be done inline instead of the defer. Relying on overwriting `err` through the whole function feels fragile.